functions/src/resolve-market.ts

```diff
@@ -31,140 +41,203 @@ export const resolveMarket = functions
     ) => {
       const userId = context?.auth?.uid
       if (!userId) return { status: 'error', message: 'Not authorized' }
-
-      const { outcome, contractId, probabilityInt, resolutions, value } = data
-
-      const contractDoc = firestore.doc(`contracts/${contractId}`)
+      const contractDoc = firestore.doc(`contracts/${data.contractId}`)
       const contractSnap = await contractDoc.get()
       if (!contractSnap.exists)
         return { status: 'error', message: 'Invalid contract' }
       const contract = contractSnap.data() as Contract
-      const { creatorId, outcomeType, closeTime } = contract
-
-      if (outcomeType === 'BINARY') {
-        if (!RESOLUTIONS.includes(outcome))
-          return { status: 'error', message: 'Invalid outcome' }
-      } else if (outcomeType === 'FREE_RESPONSE') {
-        if (
-          isNaN(+outcome) &&
-          !(outcome === 'MKT' && resolutions) &&
-          outcome !== 'CANCEL'
-        )
-          return { status: 'error', message: 'Invalid outcome' }
-      } else if (outcomeType === 'NUMERIC') {
-        if (isNaN(+outcome) && outcome !== 'CANCEL')
-          return { status: 'error', message: 'Invalid outcome' }
-      } else {
-        return { status: 'error', message: 'Invalid contract outcomeType' }
-      }
-
-      if (value !== undefined && !isFinite(value))
-        return { status: 'error', message: 'Invalid value' }
+      if (contract.creatorId !== userId)
+        return { status: 'error', message: 'User not creator of contract' }
 
+      return privateResolveMarket(contract, data)
+    }
+  )
+
+export const autoResolveMarkets = functions.pubsub
+  .schedule('every 1 minutes')
+  .onRun(async () => {
+    const contracts = await getValues<Contract>(
+      firestore
+        .collection('contracts')
+        .where('isResolved', '==', false)
+        .where('closeTime', '>', Date.now())
+        .where('autoResolutionTime', '<', Date.now())
```

Comment:
Firebase does not allow two queries with not equal operators. Advice is to move the closeTime query into a filter operation after loading the contracts.